src/dotnet/APIView/APIViewWeb/Repositories/PullRequestManager.cs

```diff
@@ -0,0 +1,182 @@
+                }                
+            }
+            catch (Exception ex)
+            {
+                telemetryClient.TrackException(ex);
+            }
+            finally
+            {
+                telemetryClient.StopOperation(operation);
+            }
+        }
+
+        // Make sure we are not processing same commit by repeated run of build pipeline
+        private async Task<bool> IsShaAlreadyProcessed(string language, int prNumber, string commitSha)
+        {
+            var pullRequestModel = await _pullRequestsRepository.GetPullRequestAsync(prNumber, language, commitSha);
+            return pullRequestModel != null;
+        }
+
+        private async Task<Stream> DownloadPackageArtifact(string buildId, string artifactName, string filePath, TelemetryClient telemetryClient)
+        {
+            try
+            {
+                var artifactGetReq = ARTIFACT_GET_URL.Replace("{buildId}", buildId).Replace("{artifactName}", artifactName);
+                var response = await devopsClient.GetAsync(artifactGetReq);
+                response.EnsureSuccessStatusCode();
+                var buildResource = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
+                if (buildResource == null)
+                {
+                    telemetryClient.TrackTrace("Artifact get response is invalid. Check REST api to get devops artifacts for any changes.");
+                    return null;
+                }
+                var downloadUrl = buildResource.RootElement.GetProperty("resource").GetProperty("downloadUrl").GetString();
+                if (!string.IsNullOrEmpty(downloadUrl))
+                {
+                    downloadUrl = downloadUrl.Split("?")[0] + "?format=file&subPath=" + filePath;
+                    var downloadResp = await devopsClient.GetAsync(downloadUrl);
+                    downloadResp.EnsureSuccessStatusCode();
+                    return await downloadResp.Content.ReadAsStreamAsync();
+                }
+                else
+                {
+                    telemetryClient.TrackTrace("Artifact get response does not have download URL. Check REST api response to get devops artifacts for any changes.");
+                    return null;
+                }
+            }
+            catch (Exception ex)
+            {
+                telemetryClient.TrackException(ex);
+            }
+            return null;
+        }
+
+        private async void UpdatedPullRequest(string diff, int prNumber, string language, TelemetryClient telemetryClient)
+        {
+            var repoName = GetGitHubRepoForLanguage(language);
+            if (repoName != "")
+            {
+                try
+                {
+                    // We should add handling of GH rate limiting here in next revision
+                    await githubClient.Issue.Comment.Create("azure", repoName, prNumber, diff);
```

Comment:
We should be careful about hardcoding this to "azure", when we start to support other scenarios we likely will need other github org support.

Reply:
Done. Changed it to configuration that can take different value based on repo name